### PR TITLE
[12.0] REST Refactor

### PR DESCRIPTION
**Pull request by dreispt**
<s>Depends on #45</s>
Changes the way taxes are computed, but only when using the REST API.

### CODE REVIEW

**Comment by dreispt** [comment]
@atchuthan I now feel that this is mature enough to merge.
I worry that the legacy  SOAP API mode is unaffected and continues working as before., but I would like independent confirmation.
It would be nice if you could test this, and give a green light for the merge.
Thanks!

**Comment by atchuthan** [comment]
@dreispt we are testing it

**Comment by dreispt** [comment]
@atchuthan Thank you. Remember to also confirm that it still works an before, when choosing SOAP mode.

**Comment by dreispt** [comment]
@atchuthan I did more tests on my end, and the SOAP mode seems to be working properly.
This should be merged, please take extra care if upgrading any of your current customers.
In case a rollback is needed, the previous commit to checkout from is 1c0dfa6.

File: avatax_connector/models/sale_order.py [hunk 1]
```diff
@@ -6,178 +6,368 @@
...
+            if order.exemption_code_id or order.exemption_code:
+                order_new_invoices = invoices & order.invoice_ids
+                order_new_invoices.write({
+                    "exemption_code": order.exemption_code or "",
+                    "exemption_code_id": order.exemption_code_id.id or False,
+                    "exemption_locked": True,
+                })
+        return invoice_ids
+
+    @api.depends("order_line.price_total", "order_line.tax_amt")
```
Comment (atchuthan) [inline avatax_connector/models/sale_order.py]: If we remove the tax selected in SO line and save it, the Taxes is not set to 0.  So, we could add Taxes field(tax_id) in SO line to depends.

**Review by atchuthan — CHANGES_REQUESTED** [review]

**Comment by atchuthan** [comment]
Also, we could add fiscal position, to set the initial tax in SO line with AVATAX record like we did it for v13(https://github.com/OCA/account-fiscal-rule/blob/13.0/account_avatax/data/avalara_salestax_data.xml)

File: avatax_connector/models/sale_order.py (hunk 1 above)
Comment (dreispt) [reply avatax_connector/models/sale_order.py]: OK, let me check that

**Comment by dreispt** [comment]
> Also, we could add fiscal position, to set the initial tax in SO line with AVATAX record like we did it for v13(https://github.com/OCA/account-fiscal-rule/blob/13.0/account_avatax/data/avalara_salestax_data.xml)

This is a significant usage change and I worry that it could cause issues on companies currently using the connector.
It would be best done in a separate PR, if you think it is important and can be done safely.

**Comment by atchuthan** [comment]
Okay, we will create a new PR for fiscal position changes in the future

File: avatax_connector/models/sale_order.py (hunk 1 above)
Comment (dreispt) [reply avatax_connector/models/sale_order.py]: Added

**Comment by dreispt** [comment]
@atchuthan Can we merge this?
I've been maintaining the refactor fork branch for a while, and that doesn't make much sense.

**Comment by atchuthan** [comment]
We are testing it once again with the latest v12 base code

**Review by atchuthan — APPROVED** [review]

**Comment by dreispt** [comment]
🎉

**Pull request merged**